### PR TITLE
feat(core): enable man build to be dynamic

### PR DESCRIPTION
**Pull request by jrschumacher**
This PR implements the proposed documentation driven CLI approach noted in #69. It is significant the number of files changed and added, but does not add significant functional changes.

The `man.go` file does not implement tests and this work has been captured in #106

### CODE REVIEW

File: docs/static/site.webmanifest [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+    "name": "OpenTDF - tructl",
```
Comment (jakedoublev) [inline docs/static/site.webmanifest:2]:
```suggestion
    "name": "OpenTDF - otdfctl",
```

File: docs/static/site.webmanifest [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    "short_name": "OpenTDF - tructl",
```
Comment (jakedoublev) [inline docs/static/site.webmanifest:3]:
```suggestion
    "short_name": "OpenTDF - otdfctl",
```

File: hugo.yaml [hunk 3]
```diff
@@ -0,0 +1,85 @@
+baseURL: http://localhost:1313
+title: tructl - OpenTDF Control Interface
```
Comment (jakedoublev) [inline hugo.yaml:2]:
```suggestion
title: otdfctl - OpenTDF Control Interface
```

File: hugo.yaml [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    - name: OpenTDF ↗
+      url: "https://opentdf.io"
+      weight: 3
+    - name: Search
+      weight: 4
+      params:
+        type: search
+    - name: GitHub
+      weight: 5
+      url: "https://github.com/opentdf/tructl"
```
Comment (jakedoublev) [inline hugo.yaml:49]:
```suggestion
      url: "https://github.com/opentdf/otdfctl"
```

File: hugo.yaml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+      width: 40
+      height: 40
+
+  footer:
+    displayCopyright: false
+    displayPoweredBy: true
+
+  editURL:
+    enable: true
+    base: "https://github.com/opentdf/tructl/edit/main/docs/"
```
Comment (jakedoublev) [inline hugo.yaml:85]:
```suggestion
    base: "https://github.com/opentdf/otdfctl/edit/main/docs/"
```

File: pkg/man/docflags.go [hunk 6]
```diff
@@ -0,0 +1,23 @@
+package man
+
+import "fmt"
+
+type DocFlag struct {
+	Name        string `yaml:"name"`
+	Description string `yaml:"description"`
+	Shorthand   string `yaml:"shorthand"`
+	Default     string `yaml:"default"`
```
Comment (jakedoublev) [inline pkg/man/docflags.go:9]: are we not actually consuming the data type from the yaml (i.e.: https://github.com/opentdf/otdfctl/pull/76/files#diff-cd82de12ea056f2f9204e40d24c61a91322da779130e06d1605d373447c29fc2R29)? If so, I think we should remove it.

File: docs/man/man.go [hunk 7]
```diff
@@ -1,47 +1,131 @@
...
+				return m.Fr[cmd]
+			}
+		}
+	}
+
+	if _, ok := m.En[cmd]; !ok {
+		panic(fmt.Sprintf("No doc found for cmd, %s", cmd))
+	}
+
+	return m.En[cmd]
```
[the remark this answers is not in the document]
Comment (jrschumacher) [reply docs/man/man.go]: I've made the enhancements, but also added a comment explaining why we're not panicing here.

**Review by jakedoublev — DISMISSED** [review]
I think the tests are failing because `pkg/logger.go` is empty and missing a package declaration.

**Review by jakedoublev — APPROVED** [review]

**Pull request merged**